### PR TITLE
MGMT-15329: Adds pkg/coreos unit-tests

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

**Comment by nmagnezi** [comment]
/assign @danielerez 
please review and propose ideas for more UTs for this pkg

File: pkg/coreos/coreos_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("DownloadISO - fail", func() {
+		mockRelease.EXPECT().ExtractFile(machineOsImageName, fmt.Sprintf(coreOsFileName, config.CpuArchitectureX86)).Return("", errors.New("some error")).Times(1)
+		_, err := testCoreOs.DownloadISO()
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("FetchCoreOSStream - fail", func() {
```
Comment (danielerez) [inline pkg/coreos/coreos_test.go:50]: For testing the success flow we can have some test json file, for example: https://github.com/openshift/assisted-installer/blob/9df495e87ee5ed9c31c281c15ceee254424b56e6/src/assisted_installer_controller/assisted_installer_controller_test.go#L2026
But maybe it's a bit overkill for this simple func.

**Review by danielerez — COMMENTED** [review]
/lgtm

**Pull request merged**